sync pull causes DeadlineExceededException

doing a sync pull on a subscription, and if there is no message, it throws `DeadlineExceededException`.  Usually, if a pull times out, it would just return null.  Also, pull wait time interval is usually configuration, how can this be configured?

```
Caused by: com.google.api.gax.rpc.DeadlineExceededException: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED: deadline exceeded after 4.978775875s. [buffered_nanos=291856840, buffered_nanos=331252614, remote_addr=pubsub.googleapis.com/172.217.12.138:443]
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:51)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1039)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1165)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:958)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:749)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:522)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:497)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426)
	at io.grpc.internal.ClientCallImpl.access$500(ClientCallImpl.java:66)
	at io.grpc.internal.ClientCallImpl$1CloseInContext.runInContext(ClientCallImpl.java:416)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	... 6 more
```

AFAIK, the current sample no longer returns `DeadlineExceededException` when there are no messages to be pulled. It fails immediately. Updating the sample to handle the no `ackId` case. 

```sh
Exception in thread "main" com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: You have not specified an ack ID in the request.
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:47)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1133)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1277)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1038)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:808)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:564)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:534)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
		at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
		at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
		at pubsub.SubscribeSyncExample.subscribeSyncExample(SubscribeSyncExample.java:80)
		at pubsub.SubscribeSyncExample.main(SubscribeSyncExample.java:43)
Caused by: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: You have not specified an ack ID in the request.
	at io.grpc.Status.asRuntimeException(Status.java:535)
	... 10 more
```

--- a/samples/snippets/src/main/java/pubsub/SubscribeSyncExample.java
+++ b/samples/snippets/src/main/java/pubsub/SubscribeSyncExample.java
@@ -61,10 +61,10 @@ public class SubscribeSyncExample {
       // Use pullCallable().futureCall to asynchronously perform this operation.
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
 
-      // Exit the program if the pull response is empty.
+      // Discontinue the program if the pull response is empty.
       if (pullResponse.getReceivedMessagesList().isEmpty()) {
         System.out.println("No message was pulled. Exiting.");
-        System.exit(0);
+        return;
       }
 
       List<String> ackIds = new ArrayList<>();

--- a/samples/snippets/src/main/java/pubsub/SubscribeSyncExample.java
+++ b/samples/snippets/src/main/java/pubsub/SubscribeSyncExample.java
@@ -60,12 +60,20 @@ public class SubscribeSyncExample {
 
       // Use pullCallable().futureCall to asynchronously perform this operation.
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
+
+      // Exit the program if the pull response is empty.
+      if (pullResponse.getReceivedMessagesList().isEmpty()) {
+        System.out.println("No message was pulled. Exiting.");
+        System.exit(0);
+      }
+
       List<String> ackIds = new ArrayList<>();
       for (ReceivedMessage message : pullResponse.getReceivedMessagesList()) {
         // Handle received message
         // ...
         ackIds.add(message.getAckId());
       }
+
       // Acknowledge received messages.
       AcknowledgeRequest acknowledgeRequest =
           AcknowledgeRequest.newBuilder()

--- a/samples/snippets/src/main/java/pubsub/SubscribeSyncExample.java
+++ b/samples/snippets/src/main/java/pubsub/SubscribeSyncExample.java
@@ -61,7 +61,8 @@ public class SubscribeSyncExample {
       // Use pullCallable().futureCall to asynchronously perform this operation.
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
 
-      // Discontinue the program if the pull response is empty.
+      // Stop the program if the pull response is empty to avoid acknowledging
+      // an empty list of ack IDs.
       if (pullResponse.getReceivedMessagesList().isEmpty()) {
         System.out.println("No message was pulled. Exiting.");
         return;

--- a/samples/snippets/src/main/java/pubsub/SubscribeSyncWithLeaseExample.java
+++ b/samples/snippets/src/main/java/pubsub/SubscribeSyncWithLeaseExample.java
@@ -38,9 +38,6 @@ public class SubscribeSyncWithLeaseExample {
     String subscriptionId = "your-subscription-id";
     Integer numOfMessages = 10;
 
-    projectId = "tz-playground-bigdata";
-    subscriptionId = "uno";
-
     subscribeSyncWithLeaseExample(projectId, subscriptionId, numOfMessages);
   }
 
@@ -68,8 +65,13 @@ public class SubscribeSyncWithLeaseExample {
       // Use pullCallable().futureCall to asynchronously perform this operation.
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
 
-      List<String> ackIds = new ArrayList<>();
+      // Exit the program if the pull response is empty.
+      if (pullResponse.getReceivedMessagesList().isEmpty()) {
+        System.out.println("No message was pulled. Exiting.");
+        System.exit(0);
+      }
 
+      List<String> ackIds = new ArrayList<>();
       for (ReceivedMessage message : pullResponse.getReceivedMessagesList()) {
         ackIds.add(message.getAckId());
 

--- a/samples/snippets/src/main/java/pubsub/SubscribeSyncWithLeaseExample.java
+++ b/samples/snippets/src/main/java/pubsub/SubscribeSyncWithLeaseExample.java
@@ -65,7 +65,8 @@ public class SubscribeSyncWithLeaseExample {
       // Use pullCallable().futureCall to asynchronously perform this operation.
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
 
-      // Discontinue the program if the pull response is empty.
+      // Stop the program if the pull response is empty to avoid acknowledging
+      // an empty list of ack IDs.
       if (pullResponse.getReceivedMessagesList().isEmpty()) {
         System.out.println("No message was pulled. Exiting.");
         return;

--- a/samples/snippets/src/main/java/pubsub/SubscribeSyncWithLeaseExample.java
+++ b/samples/snippets/src/main/java/pubsub/SubscribeSyncWithLeaseExample.java
@@ -65,10 +65,10 @@ public class SubscribeSyncWithLeaseExample {
       // Use pullCallable().futureCall to asynchronously perform this operation.
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
 
-      // Exit the program if the pull response is empty.
+      // Discontinue the program if the pull response is empty.
       if (pullResponse.getReceivedMessagesList().isEmpty()) {
         System.out.println("No message was pulled. Exiting.");
-        System.exit(0);
+        return;
       }
 
       List<String> ackIds = new ArrayList<>();

